contracts/mocks/IexecPocoMock.sol

```diff
@@ -55,4 +96,52 @@ contract IexecPocoMock is ERC20 {
     function willRevertOnSponsorMatchOrdersBoost() external {
         shouldRevertOnSponsorMatchOrdersBoost = true;
     }
+
+    function computeDealVolume(
```

Comment:
You can just return requestOrder.volume.
I think it will generally be enough.

Reply:
You can also make it public to use it in sponsorMatchOrders.

Reply:
> You can just return requestOrder.volume. I think it will generally be enough.

I rather keep it this way, since we need to check also if there is no dataset

Reply:
Changes [here](https://github.com/iExecBlockchainComputing/iexec-voucher-contracts/pull/19/commits/258d7a12d63e3db0afc9a404523340871fa3aecb)

Reply:
Reduces the diff

Reply:
> > You can just return requestOrder.volume. I think it will generally be enough.
> 
> I rather keep it this way, since we need to check also if there is no dataset

I understand but the mock is becoming complicated this way. You can achieve what you want without storing an `m_consumed`. You can return the correct value directly from orders.

Reply:
I did not check the mock carefully, I agree with Zied, we should try to keep the mock simple

Reply:
Gonna see what I can do